### PR TITLE
feat(button): CSS variable remapping

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description

Remapped the button CSS variables to match the semantic tokens

**note:** missing `d-btn-circle` remapping.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [ ] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [ ] Update, remove, or extend all affected documentation.
 - [ ] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

**Review by braddialpad — APPROVED** [review]

File: lib/build/less/components/button.less [hunk 1]
```diff
@@ -475,14 +475,14 @@
 //  $$  BRAND BUTTONS
 //  ----------------------------------------------------------------------------
 .d-btn--brand {
```
Comment (francisrupert) [inline lib/build/less/components/button.less]: No Design Token for the branded ones. Neutral white is correct value, and then the bg color is determined by the custom ones managed locally. At least for those we won't branded buttons as design tokens.

File: lib/build/less/components/button.less [hunk 2]
```diff
@@ -234,19 +233,19 @@
 .d-btn--circle {
     --button-padding-x: var(--button-padding-y-md);
     --button-padding-y: var(--button-padding-y-md);
-    --button-color-text: var(--dt-color-black-800);
+    --button-color-text: var(--dt-color-black-800); // Missing color remapping
```
Comment (francisrupert) [inline lib/build/less/components/button.less]: `action.color.foreground.muted.default`

File: lib/build/less/components/button.less [hunk 3]
```diff
@@ -234,19 +233,19 @@
...
     --button-border-radius: var(--dt-size-radius-circle);
 
     &:hover:not([disabled]) {
-        --button-color-text: var(--muted-color-hover);
-        --button-color-background: hsla(var(--dt-color-black-800-hsl) ~' / ' 3%);
+        --button-color-text: var(--muted-color-hover); // Missing color remapping
```
Comment (francisrupert) [inline lib/build/less/components/button.less]: `action.color.foreground.muted.hover`

File: lib/build/less/components/button.less [hunk 4]
```diff
@@ -234,19 +233,19 @@
...
+        --button-color-background: hsla(var(--dt-color-black-800-hsl) ~' / ' 3%); // Missing color remapping
```
Comment (francisrupert) [inline lib/build/less/components/button.less]: `action.color.background.muted.hover`

File: lib/build/less/components/button.less [hunk 5]
```diff
@@ -234,19 +233,19 @@
...
     }
 
     &:active:not([disabled]),
     &.d-btn--active:not([disabled]),
     &.d-btn--active:active:not([disabled]) {
-        --button-color-text: var(--dt-color-foreground-secondary);
-        --button-color-background: hsla(var(--dt-color-black-800-hsl) ~' / ' 9%);
+        --button-color-text: var(--dt-color-foreground-secondary); // Missing color remapping
```
Comment (francisrupert) [inline lib/build/less/components/button.less]: `action.color.background.muted.active`

File: lib/build/less/components/button.less [hunk 6]
```diff
@@ -234,19 +233,19 @@
...
+        --button-color-background: hsla(var(--dt-color-black-800-hsl) ~' / ' 9%); // Missing color remapping
```
Comment (francisrupert) [inline lib/build/less/components/button.less]: `action.color.background.muted.active`

File: lib/build/less/components/button.less [hunk 7]
```diff
@@ -258,10 +257,10 @@
     }
 
     &.d-btn--outlined {
-        --button-color-border: var(--dt-color-black-600);
+        --button-color-border: var(--dt-color-black-600); // Missing color remapping
```
Comment (francisrupert) [inline lib/build/less/components/button.less]: `action.color.border.muted.outlined.default`

**Review by francisrupert — CHANGES_REQUESTED** [review]
This is really well-done updating of local CSS variable values to `--dt-` variables.

The circle (and non-circle) icon-only ones are really just a single update, what you've already done accounts for the variants so the changes I noted are solely about the _default_ circle colors when they don't have the extra `d-btn--foo`. That default is `muted`.

**Review by francisrupert — CHANGES_REQUESTED** [review]
Ship it! Hopefully doing so for others will be more apparent. The key will be inspecting the Figma components to see what semantic styles have been applied to them. I'd hoped to create mappings for all like I did with buttons, but that's almost redundant effort.

**Comment by juliodialpad** [comment]
You meant to request changes again or you hit the wrong button and it is approved?

**Pull request merged**